llvm/lib/Target/SPIRV/SPIRVPreTranslationLegalizer.cpp

```diff
@@ -259,6 +259,12 @@ bool SPIRVPreTranslationLegalizer::runOnFunction(Function *Func,
       InitInst->setArgOperand(1, Init);
       InitConsts.insert(Init);
     }
+    if ((!GV.hasInitializer() || isa<UndefValue>(GV.getInitializer())) &&
```

Comment:
By the way why do you check that a GV does not have init value before creating call to spv_unref_global? I suppose referenced GV may be initialized (see transcoding/global-constant-expression.ll).

Reply:
Need to double-check, but I suppose initialized unreferenced globals may be already handled by the 'global_init" intrinsic.